packages/fiori/src/BarcodeScannerDialog.ts

```diff
@@ -1,22 +1,29 @@
 import UI5Element from "@ui5/webcomponents-base/dist/UI5Element.js";
 import litRender from "@ui5/webcomponents-base/dist/renderer/LitRenderer.js";
 import { getI18nBundle } from "@ui5/webcomponents-base/dist/i18nBundle.js";
+import type I18nBundle from "@ui5/webcomponents-base/dist/i18nBundle.js";
 import Dialog from "@ui5/webcomponents/dist/Dialog.js";
 import Button from "@ui5/webcomponents/dist/Button.js";
 import BusyIndicator from "@ui5/webcomponents/dist/BusyIndicator.js";
 import * as ZXing from "@zxing/library/umd/index.min.js";
-// Template
-import BarcodeScannerDialogTemplate from "./generated/templates/BarcodeScannerDialogTemplate.lit.js";
-
-// Styles
-import BarcodeScannerDialogCss from "./generated/themes/BarcodeScannerDialog.css.js";
+import customElement from "@ui5/webcomponents-base/dist/decorators/customElement.js";
+import languageAware from "@ui5/webcomponents-base/dist/decorators/languageAware.js";
+import property from "@ui5/webcomponents-base/dist/decorators/property.js";
+import event from "@ui5/webcomponents-base/dist/decorators/event.js";
+import type { BrowserMultiFormatReader as BrowserMultiFormatReaderT, Result, Exception } from "@zxing/library/esm5/index";
 
 // Texts
 import {
 	BARCODE_SCANNER_DIALOG_CANCEL_BUTTON_TXT,
 	BARCODE_SCANNER_DIALOG_LOADING_TXT,
 } from "./generated/i18n/i18n-defaults.js";
 
+// Template
+import BarcodeScannerDialogTemplate from "./generated/templates/BarcodeScannerDialogTemplate.lit.js";
+
+// Styles
+import BarcodeScannerDialogCss from "./generated/themes/BarcodeScannerDialog.css.js";
+
 // some tools handle named exports from UMD files and the window object is not assigned but the imports work (vitejs)
 // other tools do not handle named exports (they are undefined after the import), but the window global is assigned and can be used (web dev server)
 const effectiveZXing = { ...ZXing, ...window.ZXing };
```

Comment:
The code looks good, no +1 still, because there is just one thing that has to be discussed and fixed. 
As it is specific, our team is looking at and it's related to the ZXing types.

If we remove the ` ...window.ZXing` part and leave it like this:
```ts
const effectiveZXing = { ...ZXing };
```
We noticed that the ZXing internals types evaluate  to "any" type which is not correct. 
The root cause is also known - there are some changes to be done in the `BarcodeScannerDialog-zxing.d.ts` file.
We will keep you updated once the solution is ready.